### PR TITLE
[ZEPPELIN-6080] Add support for bq single region dataset query

### PR DESCRIPTION
**Pull request by meharanjan318**
### What is this PR for?
A few sentences describing the overall goals of the pull request's commits.
First time? Check out the contributing guide - https://zeppelin.apache.org/contribution/contributions.html

### What type of PR is it?
Improvement
*Please leave your type of PR only*

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-6080

### How should this be tested?
*CI
 local tests

### Questions:
* Does the license files need to update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes

### CODE REVIEW

**Comment by meharanjan318** [comment]
@pan3793 can you please take a look.

File: bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java [hunk 1]
```diff
@@ -256,6 +263,33 @@ private InterpreterResult executeSql(String sql) {
     }
   }
 
+  //Function to run the SQL on bigQuery service
+  public static Iterator<GetQueryResultsResponse> run(final String queryString,
```
[the remark this answers is not in the document]
Comment (meharanjan318) [reply bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java:265]: Done.

File: bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java [hunk 2]
```diff
@@ -90,9 +90,13 @@ public class BigQueryInterpreter extends Interpreter {
   static final String WAIT_TIME = "zeppelin.bigquery.wait_time";
   static final String MAX_ROWS = "zeppelin.bigquery.max_no_of_rows";
   static final String SQL_DIALECT = "zeppelin.bigquery.sql_dialect";
+  static final String REGION = "zeppelin.bigquery.region";
+
```
Comment (Reamer) [inline bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java:94]: nit - Please remove this line.

File: bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java [hunk 3]
```diff
@@ -275,6 +281,7 @@ public static Iterator<GetQueryResultsResponse> run(final String queryString,
       GetQueryResults getRequest = service.jobs().getQueryResults(
           projectId,
           jobId);
+      if (region != null) getRequest = getRequest.setLocation(region);
```
Comment (Reamer) [inline bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java:284]: only a style change
```suggestion
      if (region != null) {
         getRequest = getRequest.setLocation(region);
      }
```

File: bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java [hunk 4]
```diff
@@ -90,9 +90,13 @@ public class BigQueryInterpreter extends Interpreter {
   static final String WAIT_TIME = "zeppelin.bigquery.wait_time";
   static final String MAX_ROWS = "zeppelin.bigquery.max_no_of_rows";
   static final String SQL_DIALECT = "zeppelin.bigquery.sql_dialect";
+  static final String REGION = "zeppelin.bigquery.region";
+
 
   private static String jobId = null;
   private static String projectId = null;
+  private static String location = null;
```
Comment (Reamer) [inline bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java:98]: I don't see any use for this new variable.

**Review by Reamer — CHANGES_REQUESTED** [review]
Much better, just small things

File: bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java (hunk 2 above)
Comment (meharanjan318) [reply bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java:94]: Done.

File: bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java (hunk 4 above)
Comment (meharanjan318) [reply bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java]: Done.

File: bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java (hunk 3 above)
Comment (meharanjan318) [reply bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java]: Done.

**Review by Reamer — APPROVED** [review]
Code changes are looking good to me. Just the Pull request description should be updated.

**Review by jongyoul — DISMISSED** [review]
LGTM except for the description @Reamer already mentioned.

**Comment by meharanjan318** [comment]
@Reamer @jongyoul updated the PR description and documentation file.

**Review by Reamer — APPROVED** [review]
LGTM

**Comment by meharanjan318** [comment]
@Reamer can you please merge the PR as well if it looks good.

File: bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java [hunk 5]
```diff
@@ -275,6 +278,9 @@ public static Iterator<GetQueryResultsResponse> run(final String queryString,
       GetQueryResults getRequest = service.jobs().getQueryResults(
           projectId,
           jobId);
+      if (region != null) {
```
Comment (Reamer) [inline bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java:281]: This check should be better. Not that somehow an empty string or a string with white spaces arrives here. Please consider the import of the function. It is best to also include `commons-lang3` in the pom.xml.
Sorry about the back and forth.
```suggestion
      if (StringUtils.isNotBlank(region)) {
```

**Review by Reamer — CHANGES_REQUESTED** [review]

File: bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java (hunk 5 above)
Comment (meharanjan318) [reply bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryInterpreter.java]: Done

**Review by Reamer — APPROVED** [review]
Thanks, LGTM

**Pull request merged**